aiohttp/client.py

```diff
@@ -667,17 +675,40 @@ async def _request(
                         max_field_size=max_field_size,
                     )
 
-                    try:
+                    # Core request handler - use default parameter to avoid loop variable issues
+                    async def _send_request(
+                        req: ClientRequest, *, _conn: Connection = conn
+                    ) -> ClientResponse:
                         try:
-                            resp = await req.send(conn)
+                            resp = await req.send(_conn)
                             try:
-                                await resp.start(conn)
+                                await resp.start(_conn)
                             except BaseException:
                                 resp.close()
                                 raise
                         except BaseException:
-                            conn.close()
+                            _conn.close()
                             raise
+
+                        return resp
```

Comment:
This pattern is a problem because if the middleware raises we won't close the connection. We need to remove the outer try and instead catch it below